ext/dom/parentnode.c

```diff
@@ -528,95 +751,80 @@ void dom_child_node_remove(dom_object *context)
 	xmlUnlinkNode(child);
 }
 
+/* https://dom.spec.whatwg.org/#dom-childnode-replacewith */
 void dom_child_replace_with(dom_object *context, zval *nodes, uint32_t nodesc)
 {
-	/* Spec link: https://dom.spec.whatwg.org/#dom-childnode-replacewith */
+	if (UNEXPECTED(dom_sanity_check_node_list_types(nodes, nodesc, dom_get_node_ce(php_dom_follow_spec_doc_ref(context->document))) != SUCCESS)) {
+		return;
+	}
 
 	xmlNodePtr child = dom_object_get_node(context);
 
-	/* Spec step 1 */
+	/* 1. Let parent be this’s parent. */
 	xmlNodePtr parentNode = child->parent;
 
-	/* Sanity check for fragment, includes spec step 2 */
-	if (UNEXPECTED(dom_sanity_check_node_list_for_insertion(context->document, parentNode, nodes, nodesc) != SUCCESS)) {
-		return;
-	}
-
-	int stricterror = dom_get_strict_error(context->document);
-	if (UNEXPECTED(dom_child_removal_preconditions(child, stricterror) != SUCCESS)) {
+	/* 2. If parent is null, then return. */
+	if (UNEXPECTED(parentNode == NULL)) {
 		return;
 	}
 
-	/* Spec step 3: find first following child not in nodes; otherwise null */
+	/* 3. Let viableNextSibling be this’s first following sibling not in nodes; otherwise null. */
 	xmlNodePtr viable_next_sibling = child->next;
-	while (viable_next_sibling) {
-		if (!dom_is_node_in_list(nodes, nodesc, viable_next_sibling)) {
-			break;
-		}
+	while (viable_next_sibling && dom_is_node_in_list(nodes, nodesc, viable_next_sibling)) {
 		viable_next_sibling = viable_next_sibling->next;
 	}
 
-	xmlDocPtr doc = parentNode->doc;
-	php_libxml_invalidate_node_list_cache(context->document);
-
-	/* Spec step 4: convert nodes into fragment */
-	xmlNodePtr fragment = dom_zvals_to_fragment(context->document, parentNode, nodes, nodesc);
-	if (UNEXPECTED(fragment == NULL)) {
+	int stricterror = dom_get_strict_error(context->document);
```

Comment:
Nit: (although I'm guessing this is old DOM code)
```suggestion
	bool stricterror = dom_get_strict_error(context->document);
```